src/Core/Services/StateService.cs

```diff
@@ -1720,6 +1720,22 @@ public async Task<bool> GetLastUserShouldConnectToWatchAsync()
             return await GetValueAsync<bool?>(key, options) ?? false;
         }
 
+        public async Task SetAvatarColorAsync(string value, string userId = null)
+        {
+            var reconciledOptions = ReconcileOptions(new StorageOptions { UserId = userId },
+                await GetDefaultStorageOptionsAsync());
+            var account = await GetAccountAsync(reconciledOptions);
```

Comment:
Establish a key and use `SetValueAsync` here instead of working directly on the `account` object (check the other setters for reference - The account object is automatically handled by setting the value).